app/code/Magento/Catalog/Controller/Adminhtml/Product/MassDelete.php

```diff
@@ -54,16 +59,23 @@ public function __construct(
     }
 
     /**
+     * Mass Delete Action
+     *
      * @return \Magento\Backend\Model\View\Result\Redirect
      */
     public function execute()
     {
         $collection = $this->filter->getCollection($this->collectionFactory->create());
         $productDeleted = 0;
+        $productDeletedError = 0;
         /** @var \Magento\Catalog\Model\Product $product */
         foreach ($collection->getItems() as $product) {
-            $this->productRepository->delete($product);
-            $productDeleted++;
+            try {
+                $this->productRepository->delete($product);
+                $productDeleted++;
+            } catch (\Exception $exception) {
```

Comment:
I think this is not the best practice to catch `\Exception` in general.
According to the service contract of `\Magento\Catalog\Api\ProductRepositoryInterface`, the `delete` method throws `\Magento\Framework\Exception\StateException`. 
And, according to the realization, repository throws `\Magento\Framework\Exception\CouldNotSaveException`.
Both can be generalized as `\Magento\Framework\Exception\LocalizedException`.